packages/shared/src/components/comments/CommentInputPage.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { ReactElement, useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { useRouter } from 'next/router';
+import { FormWrapper } from '../fields/form';
+import { CommentMarkdownInput } from '../fields/MarkdownInput/CommentMarkdownInput';
+import { useMutateComment } from '../../hooks/post/useMutateComment';
+import { useVisualViewport } from '../../hooks/utils/useVisualViewport';
+import { COMMENT_BY_ID_WITH_POST_QUERY } from '../../graphql/comments';
+import useCommentById from '../../hooks/comments/useCommentById';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { Switch } from '../fields/Switch';
+import { useNotificationToggle } from '../../hooks/notifications';
+import { NotificationPromptSource } from '../../lib/analytics';
+import { Post } from '../../graphql/posts';
+
+// statically defined heights of various containers that
+// affect final height of the input container
+// when styles change, these should be updated accordingly
+const headerSize = 57;
+const replySize = 40;
+const footerSize = 48;
+export interface Props {
+  postId: string;
+  editCommentId?: string;
+  replyCommentId?: string;
+}
+
+const CommentInputPage = ({
+  postId,
+  editCommentId,
+  replyCommentId,
+}: Props): ReactElement => {
+  const [value, setValue] = useState('');
+  const [styleHeight, setStyleHeight] = useState<number | 'auto'>('auto');
+  const { user } = useAuthContext();
+  const router = useRouter();
+
+  const { shouldShowCta, isEnabled, onToggle, onSubmitted } =
+    useNotificationToggle({
+      source: NotificationPromptSource.NewComment,
+    });
+
+  const isEdit = !!editCommentId;
+  const isReply = !!replyCommentId;
+
+  const { comment } = useCommentById({
+    id: isEdit ? editCommentId : replyCommentId,
+    query: COMMENT_BY_ID_WITH_POST_QUERY,
+    options: {
+      retry: false,
+      enabled: !!editCommentId || !!replyCommentId,
+    },
+  });
+  const post = useMemo(
+    () => comment?.post ?? ({ id: postId } as Post),
+    [comment, postId],
+  );
+  const parentCommentId = useMemo(
+    () => comment?.parent?.id ?? replyCommentId,
+    [comment, replyCommentId],
+  );
+
+  const { mutateComment, isLoading } = useMutateComment({
+    post,
+    editCommentId: isEdit && editCommentId,
+    parentCommentId,
+    onCommented: router.back,
+  });
+
+  const { height } = useVisualViewport();
+  const replyHeight = height > 0 ? replySize : 0;
+  const footerHeight = shouldShowCta ? footerSize : 0;
+  const totalHeight = height - headerSize - replyHeight - footerHeight;
+  const inputHeight = totalHeight > 0 ? totalHeight : 'auto';
```

Comment:
this could be better in a helper function

Reply:
Created a task to track this: https://dailydotdev.atlassian.net/browse/MI-240